relayer/src/supervisor.rs

```diff
@@ -743,17 +770,27 @@ fn add_chain<Chain: ChainHandle>(
         return CmdEffect::Nothing;
     }
 
-    debug!(chain.id=%id, "spawning workers");
+    debug!(chain.id=%id, "scanning chain");
 
-    let mut ctx = spawn_context(
-        config,
-        registry,
-        client_state_filter,
-        workers,
-        SpawnMode::Reload,
-    );
+    let scan_result = chain_scanner(config, registry, client_state_filter, ScanMode::Auto)
+        .scan_chain(&chain_config);
+
+    let scan = match scan_result {
+        Ok(scan) => scan,
+        Err(e) => {
+            error!("failed to scan chain {}: {}", id, e);
 
-    ctx.spawn_workers_for_chain(&id);
+            // Remove the newly added config
+            config.chains.retain(|c| c.id != id);
+
+            return CmdEffect::Nothing;
+        }
+    };
+
+    debug!(chain.id=%id, "spawning workers");
```

Comment:
Not sure if the dots in key names are a good idea.
At least for the systemd backend, I think it's better to stick to alphanumerics and underscores.

Reply:
a414523d